src/remote/workspace.py

```diff
@@ -28,17 +28,28 @@ class SyncedWorkspace:
 
     @classmethod
     def from_config(
-        cls, config: WorkspaceConfig, working_dir: Path, config_num: Optional[int] = None
+        cls, config: WorkspaceConfig, working_dir: Path, config_num: Optional[Union[str, int]] = None
```

Comment:
let's rename `config_num` to something more reasonable, like `remote_host_id` or something else